File: app/src/main/java/com/kaltura/kflow/presentation/player/PlayerKeepAliveService.java
package com.kaltura.kflow.presentation.player;

import android.os.Handler;
import android.util.Log;

import java.net.HttpURLConnection;
import java.net.URL;

/**
 * Created by alex_lytvynenko on 2019-10-28.
 */
class PlayerKeepAliveService {

    private final static String TAG = PlayerKeepAliveService.class.getCanonicalName();

    private final long KEEP_ALIVE_CYCLE = 10000L;
    private Handler scheduler = null;
    private String keepAliveURL;

    PlayerKeepAliveService() {
        keepAliveURL = "";
        if (scheduler == null) {
            scheduler = new Handler();
        }
        scheduler.removeCallbacks(fireKeepAliveCallsRunnable);
    }

    void startFireKeepAliveService() {
        if (scheduler != null) {
            cancelFireKeepAliveService();
            fireKeepAliveHeaderUrl(keepAliveURL);
            scheduler.postDelayed(fireKeepAliveCallsRunnable, KEEP_ALIVE_CYCLE);
        }
    }

    void cancelFireKeepAliveService() {
        if (scheduler != null) {
            scheduler.removeCallbacks(fireKeepAliveCallsRunnable);
        }
    }

    void setKeepAliveURL(String url) {
        keepAliveURL = url;
    }

    private static void fireKeepAliveHeaderUrl(final String url) {
        new Thread(() -> {
            try {
                String keepAliveStr = url + "/" + "keepalive";
                URL firedKAURL = new URL(keepAliveStr);
                HttpURLConnection conn = (HttpURLConnection) firedKAURL.openConnection();
                conn.setInstanceFollowRedirects(false);
                final boolean isSuccess = conn.getResponseCode() == 204;
                if (isSuccess) {
                    Log.d(TAG, "Firing KeepAliveURL : " + firedKAURL);
                } else {
                    Log.d(TAG, "Firing KeepAliveURL failed : " + firedKAURL);
                }

            } catch (Exception e) {
                e.printStackTrace();
            }

        }).start();
    }

    private Runnable fireKeepAliveCallsRunnable = () -> {
        startFireKeepAliveService();
    };
}

File: app/src/main/java/com/kaltura/kflow/presentation/player/PlayerKeepAliveService.kt
package com.kaltura.kflow.presentation.player

import android.os.Handler
import android.util.Log
import java.net.HttpURLConnection
import java.net.URL

/**
 * Created by alex_lytvynenko on 2019-10-28.
 */
class PlayerKeepAliveService {
    private val TAG = PlayerKeepAliveService::class.java.canonicalName

    private val KEEP_ALIVE_CYCLE = 10000L
    private var scheduler = Handler()
    private val fireKeepAliveCallsRunnable = Runnable { startFireKeepAliveService() }
    var keepAliveURL = ""

    init {
        scheduler.removeCallbacks(fireKeepAliveCallsRunnable)
    }

    fun startFireKeepAliveService() {
        cancelFireKeepAliveService()
        fireKeepAliveHeaderUrl(keepAliveURL)
        scheduler.postDelayed(fireKeepAliveCallsRunnable, KEEP_ALIVE_CYCLE)
    }

    fun cancelFireKeepAliveService() {
        scheduler.removeCallbacks(fireKeepAliveCallsRunnable)
    }

    private fun fireKeepAliveHeaderUrl(url: String) {
        Thread(Runnable {
            try {
                val keepAliveStr = "$url/keepalive"
                val firedKAURL = URL(keepAliveStr)
                val conn = firedKAURL.openConnection() as HttpURLConnection
                conn.instanceFollowRedirects = false
                val isSuccess = conn.responseCode == 204

                if (isSuccess) Log.d(TAG, "Firing KeepAliveURL : $firedKAURL")
                else Log.d(TAG, "Firing KeepAliveURL failed : $firedKAURL")

            } catch (e: Exception) {
                e.printStackTrace()
            }
        }).start()
    }
}